src/bun.js/bindings/bindings.cpp

```diff
@@ -1371,7 +1375,11 @@ std::optional<bool> specialObjectsDequal(JSC__JSGlobalObject* globalObject, Mark
     compareAsNormalValue:
         break;
     }
-
+    // globalThis is only equal to globalThis
+    // NOTE: Zig::GlobalObject is tagged as GlobalProxyType
+    case GlobalObjectType:
+    case GlobalProxyType:
+        return c1Type == c2Type;
```

Comment:
This is not correct. A node:vm JSGlobalObject/globalThis could be compared to here. It should check if the values are strict equal for that case.

Reply:
I think I've fixed it, but I'm not sure. Will confirm and update afterwards.